torchbox/torchbox.com
# QuoteBlock BE

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/views/6453316?term=quot)

### Description of Changes Made
This work builds the backend to complement @chris-lawton 's work on the [frontend](https://github.com/torchbox/torchbox.com/pull/26)

- I've included additional company_logo, author_image and link fields on the PullQuoteBlock.
- StructBlockValidation is used to limit editors to choose either a company_logo or an author_image but not both.
- These errors are raised on the specific error fields of the admin as well as at the top of the page
- Each of these fields are optional
- The link field makes use of the existing, unused LinkBlock()

### How to Test
- Create a QuoteBlock on any page
- Include a quote & attribution, a link and either an author_image or company_logo. The page should save successfully.
- Show that a validation error is raised if both the author_image and company_logo are created.
- Show that all of author_image, company_logo and link are optional
- The frontend should render as in the figma (I've integrated the FE changes now)

### Screenshots

<details>
  <summary>Expand to see more</summary>

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
